build_tools/python/benchmark_suites/iree/x86_64_benchmarks.py

```diff
@@ -40,26 +40,44 @@ def _generate_default(
       self, device_specs: List[common_definitions.DeviceSpec]
   ) -> Tuple[List[iree_definitions.ModuleGenerationConfig],
              List[iree_definitions.E2EModelRunConfig]]:
-    gen_configs = [
+    # Create benchmarks for small workloads.
+    gen_configs_small = [
         iree_definitions.ModuleGenerationConfig.build(
             compile_config=self.CASCADELAKE_COMPILE_CONFIG,
             imported_model=iree_definitions.ImportedModel.from_model(model))
-        for model in model_groups.SMALL + model_groups.LARGE
+        for model in model_groups.SMALL
     ]
-
-    default_execution_configs = [
-        module_execution_configs.ELF_LOCAL_SYNC_CONFIG
-    ] + [
-        module_execution_configs.get_elf_local_task_config(thread_num)
-        for thread_num in [1, 4, 8]
+    execution_configs_small = [
+        module_execution_configs.get_elf_local_task_config(1),
+        module_execution_configs.get_elf_local_task_config(8)
     ]
+    run_configs_small = benchmark_suites.iree.utils.generate_e2e_model_run_configs(
+        module_generation_configs=gen_configs_small,
+        module_execution_configs=execution_configs_small,
+        device_specs=device_specs)
 
-    run_configs = benchmark_suites.iree.utils.generate_e2e_model_run_configs(
-        module_generation_configs=gen_configs,
-        module_execution_configs=default_execution_configs,
+    # Create benchmarks for large workloads.
+    gen_configs_large = [
+        iree_definitions.ModuleGenerationConfig.build(
+            compile_config=self.CASCADELAKE_COMPILE_CONFIG,
+            imported_model=iree_definitions.ImportedModel.from_model(model))
+        for model in model_groups.LARGE
+    ]
+    # We use higher thread counts for large workloads.
+    execution_configs_large = [
+        module_execution_configs.get_elf_local_task_config(1),
+        module_execution_configs.get_elf_local_task_config(8),
+        module_execution_configs.get_elf_local_task_config(16),
+        # We use (MAX_CORE - 2) here to determine the maximum number of threads to use on a NUMA node.
+        module_execution_configs.get_elf_local_task_config(28),
```

Comment:
Nit: for future improvements maybe a good idea to get max cpu count from `numactl --hardware` or some python wrapper for it so it should work whichever VM you deploy on.  Also maybe good to document the two cores are used by IREE for the scheduling (and last I tested they had to be on numa node 0).  

For peak performance on cpu we also need to enable the spin wait scheduling that Ben had added so we don't go wake each thread up as that adds up for 32 threads.

Reply:
This is the benchmark suite to track regressions so we kinda want to have well-defined and easy to understand benchmarks instead of doing something too smart behind the scene. If we change/add new machines, it will be defined in the benchmark suites and we define new benchmarks for them.

We also want to at least track with the default flags. So if there are some special tuning, they should be defined as special (experimental) benchmarks.